csharp/src/IceLocatorDiscovery/PluginI.cs

```diff
@@ -457,8 +409,11 @@ public void
             }
         }
 
-        async Task preformFindLocatorAsync(LookupPrx lookupPrx, LookupReplyPrx lookupReplyPrx)
+        async Task performFindLocatorAsync(LookupPrx lookupPrx, LookupReplyPrx lookupReplyPrx)
         {
+            // Exit the mutex lock before proceeding.
```

Comment:
Awaiting a remote call with a mutex lock (see below) is incorrect and deadlock prone. It could be the cause for #2813.